gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsRecord.java

```diff
@@ -0,0 +1,270 @@
+     * non-negative.
+     *
+     * @param ttl A non-negative number of seconds
+     */
+    public Builder ttl(int ttl) {
+      checkArgument(ttl >= 0, "TTL cannot be negative. The supplied value was " + ttl + ".");
+      this.ttl = ttl;
+      return this;
+    }
+
+    /**
+     * The identifier of a supported record type, for example, A, AAAA, MX, TXT, and so on.
+     */
+    public Builder type(DnsRecordType type) {
+      this.type = checkNotNull(type);
+      return this;
+    }
+
+    /**
+     * Builds the DNS record.
+     */
+    public DnsRecord build() {
+      return new DnsRecord(this);
+    }
+
+    /**
+     * Sets references to the managed zone that this DNS record belongs to.
+     */
+    public Builder managedZone(ManagedZoneInfo parent) {
+      checkNotNull(parent);
+      this.zoneId = parent.id();
+      this.zoneName = parent.name();
+      return this;
+    }
+  }
+
+  /**
+   * Creates a builder pre-populated with the attribute values of this instance.
+   */
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  /**
+   * Creates an empty builder
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  /**
+   * Get the mandatory user assigned name of this DNS record.
+   */
+  public String name() {
+    return name;
+  }
+
+  /**
+   * Returns a list of DNS record stored in this record set.
+   */
+  public List<String> rrdatas() {
+    return ImmutableList.copyOf(rrdatas);
+  }
+
+  /**
+   * Returns the number of seconds that this ResourceRecordSet can be cached by resolvers. This
+   * number is provided by the user.
+   */
+  public Integer ttl() {
+    return ttl;
+  }
+
+  /**
+   * Returns the type of this DNS record.
+   */
+  public DnsRecordType type() {
+    return type;
+  }
+
+  /**
+   * Returns name of the managed zone that this record belongs to. The name of the managed zone is
+   * provided by the user when the managed zone is created. It is unique within a project. If this
+   * DNS record is not associated with a managed zone, this returns null.
+   */
+  public String zoneName() {
+    return zoneName;
+  }
+
+  /**
+   * Returns name of the managed zone that this record belongs to.
+   *
+   * <p> The id of the managed zone is determined by the server when the managed zone is created. It
+   * is a read only value. If this DNS record is not associated with a managed zone, or if the id of
+   * the managed zone was not loaded from the cloud service, this returns null.
+   */
+  public Long zoneId() {
+    return zoneId;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(name, rrdatas, ttl, type, zoneName, zoneId);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj instanceof DnsRecord) {
+      DnsRecord other = (DnsRecord) obj;
+      return zoneId == other.zoneId()
+              && zoneName == other.zoneName
+              && this.toRRSet().equals(other.toRRSet());
```

Comment:
Here, above and in `toRRSet` we should either use getters (`zoneId()`) or fields (`zoneId`) to access properties. And we should either qualify all (`this.`) or leave it implicit.